contract/src/agoric-basics.contract.js

```diff
@@ -40,58 +40,112 @@ const bagCounts = bag => {
   const entries = getCopyBagEntries(bag);
   return entries.map(([_k, ct]) => ct);
 };
+
+/**
+ *
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @param {Object.<string, {tradePrice: Amount, maxTickets: bigint}>} inventory
+ * @returns {boolean}
+ */
+export const hasInventory = (bag, inventory) => {
+  const entries = getCopyBagEntries(bag);
+  for (const [k, ct] of entries) {
+    if (k in inventory) {
+      const { maxTickets } = inventory[k];
+      if (ct > maxTickets) {
+        return false;
+      }
+    } else {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ *
+ * @param {Amount} amount
+ * @param {number} n
+ * @returns {Amount}
+ */
+const multiply = (amount, n) => {
+  const arr = Array.from({ length: n });
+  return arr.reduce(
+    (sum, _) => AmountMath.add(amount, sum),
+    AmountMath.make(amount.brand, 0n),
+  );
+};
+
+/**
+ *
+ * @param {Amount} sum
+ * @param {[string, bigint]} entry
+ * @param {Object.<string, {tradePrice: Amount, maxTickets: bigint}>} inventory
+ * @returns {Amount}
+ */
+const addMultiples = (sum, entry, inventory) => {
+  const multiple = multiply(inventory[entry[0]].tradePrice, Number(entry[1]));
+  return AmountMath.add(multiple, sum);
+};
+
+/**
+ *
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @param {Object.<string, {tradePrice: Amount, maxTickets: bigint}>} inventory
+ * @returns {Amount}
+ */
+export const bagPrice = (bag, inventory) => {
+  const entries = getCopyBagEntries(bag);
+  const brand = Object.values(inventory)[0].tradePrice.brand;
```

Comment:
_question_

This is a bug if inventory is empty, but I can't think of a better way to get the brand?

Reply:
indeed, `bagPrice` is not well-defined if inventory is empty. So assert that it's not:

```suggestion
  const vs = Object.values(inventory);
  vs.length > 0 || Fail`inventory must not be empty`;
  const { brand } = vs[0].tradePrice;
```

It's also not well-defined if the brands of the `tradePrice`s are not all the same. `AmountMath.add()` will catch that, though ideally it should be checked at contract start.